coreplugins/cesiumion/public/components/UploadDialog.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { Component, Fragment } from "react";
+
+import FormDialog from "../../../../app/static/app/js/components/FormDialog";
+
+import BootstrapField from "./BootstrapField";
+import { ImplicitIonFetcher as IonFetcher } from "./Fetcher";
+import { AssetType, SourceType } from "../defaults";
+import "./UploadDialog.scss";
+
+export default class UploadDialog extends Component {
+	static AssetSourceType = {
+		[AssetType.ORTHOPHOTO]: SourceType.RASTER_IMAGERY,
+		[AssetType.TERRAIN_MODEL]: SourceType.RASTER_TERRAIN,
+		[AssetType.SURFACE_MODEL]: SourceType.RASTER_TERRAIN,
+		[AssetType.POINTCLOUD]: SourceType.POINTCLOUD,
+		[AssetType.TEXTURED_MODEL]: SourceType.CAPTURE
+	};
+
+	static defaultProps = {
+		show: true,
+		asset: null,
+		loading: false,
+		initialValues: {
+			name: "",
+			description: "",
+			attribution: "",
+			options: {
+				baseTerrainId: "",
+				textureFormat: false
+			}
+		}
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.mergedInitialValues = {
+			...UploadDialog.defaultProps.initialValues,
+			...this.props.initialValues
+		};
+
+		this.state = {
+			title : props.title,
+			...this.mergedInitialValues
+		}
+	}
+
+    show(){
+        this.dialog.show();
+    }
+
+    handleChange = (e) => {
+        const { value, name } = e.target;
+		
+		if (name === "options.textureFormat")
+		{
+			let options = {...this.state.options};
+			options["textureFormat"] = value === "Yes";
+			this.setState({ options });
+		}
+		else if (name === "options.baseTerrainId")
+		{
+			let options = {...this.state.options};
+			options["baseTerrainId"] = value;
+			this.setState({ options });
+		}
+		else
+		{
+			this.setState({ [name]: value });
+		}
+    }
+
+	handleError = msg => error => {
+		this.props.onError(msg);
+		console.error(error);
+	};
+
+	onSubmit = values => {
+		const { asset, onSubmit } = this.props;
+		values = JSON.parse(JSON.stringify(this.state));
+		const { options = {} } = values;
+
+		switch (UploadDialog.AssetSourceType[asset]) {
+			case SourceType.RASTER_TERRAIN:
+				if (options.baseTerrainId === "")
+					delete options["baseTerrainId"];
+				else options.baseTerrainId = parseInt(options.baseTerrainId);
+				options.toMeters = 1;
+				options.heightReference = "WGS84";
+				options.waterMask = false;
+				break;
+			case SourceType.CAPTURE:
+				options.textureFormat = options.textureFormat ? "KTX2" : "AUTO";
+				break;
+		}
+
+		onSubmit(values);
+	};
+
+	getSourceFields() {
+		switch (UploadDialog.AssetSourceType[this.props.asset]) {
+			case SourceType.RASTER_TERRAIN:
+				let loadOptions = ({ isLoading, isError, data }) => {
+					if (isLoading || isError)
+						return <option disabled>LOADING...</option>;
+					let userItems = data.items
+						.filter(item => item.type === "TERRAIN")
```

Comment:
Can we make a "enum" for these types

Reply:
`RASTER_TERRAIN` are uploads that would be overlay-ed on the `TERRAIN` of an existing asset on cesium ion. There isn't a need to convert it to an enum. Instead I could convert the string into a `const string` at the start of the function.